File: NirvanaSMS.Api.CSharp/Core/Entity/OneToManyMessage.cs
using NirvanaSMS.Api.Core.Base;
using NirvanaSMS.Api.Core.Interface;
using NirvanaSMS.Api.Core.Utility;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace NirvanaSMS.Api.Core.Entity
{
    public class OneToManyMessage : MessageBase
    {
        public string Message { get; set; }
        public List<PhoneNumber> GsmNumbers { get; set; }
        public OneToManyMessageBuilder Builder { get; set; }

        public OneToManyMessage()
        {
            Builder = new OneToManyMessageBuilder();
        }

        public OneToManyMessage(string header, string message, string[] phoneNumbers, int validityTime)
        {
            Header = header;
            Message = message;
            GsmNumbers = phoneNumbers.Select(x => new PhoneNumber(x)).ToList();
            ValidityTime = validityTime;
        }

        public OneToManyMessage(OneToManyMessageBuilder builder)
        {
            Header = builder.Header;
            Message = builder.Message;
            GsmNumbers = builder.GsmNumbers;
            ValidityTime = builder.ValidityTime;
            SendDateTime = builder.SendDateTime;
        }
    }
}

File: NirvanaSMS.Api.Java/src/main/java/org/nirvanasms/api/core/entity/OneToManyMessage.java
package org.nirvanasms.api.core.entity;

import java.util.List;

import org.nirvanasms.api.core.base.MessageBase;
import org.nirvanasms.api.core.utility.OneToManyMessageBuilder;

public class OneToManyMessage extends MessageBase {
	private String message;
	private List<PhoneNumber> gsmNumbers;
	
	public OneToManyMessage(){}
	
	public OneToManyMessage(OneToManyMessageBuilder builder){
		super.setHeader(builder.getHeader());
		this.message = builder.getMessage();
		this.gsmNumbers = builder.getGsmNumbers();
		super.setValidityTime(builder.getValidityTime());
		super.setSendDateTime(builder.getSendDateTime());
	}
	public String getMessage() {
		return message;
	}
	public void setMessage(String message) {
		this.message = message;
	}
	public List<PhoneNumber> getGsmNumbers() {
		return gsmNumbers;
	}
	public void setGsmNumbers(List<PhoneNumber> gsmNumbers) {
		this.gsmNumbers = gsmNumbers;
	}
	
	public OneToManyMessageBuilder builder(){
		return new OneToManyMessageBuilder();
	}
}
